### PR TITLE
Reader: Use redux state for feeds and sites in full post

### PR DESCRIPTION
**Pull request by blowery**
Piggy-backs on how we're doing this for new style streams.

No real functional changes, but fewer network requests if you're using the refreshed streams.

### CODE REVIEW

**Comment by blowery** [comment]
Fixes #8797

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -353,14 +355,38 @@ export class FullPostView extends React.Component {
 	}
 }
 
+const ConnectedFullPostView = connect(
+	( state, ownProps ) => {
+		const {
+			site_ID: siteId,
+			feed_ID: feedId,
+			isExternal: is_external
```
Comment (bluefuton) [inline client/blocks/reader-full-post/index.jsx]: Is this the wrong way round? Looks like we should be mapping `is_external` on the `post` to the `isExternal` const.

Comment (blowery) [inline client/blocks/reader-full-post/index.jsx]: Yes! Good catch.

Comment (blowery) [inline client/blocks/reader-full-post/index.jsx]: Yes! Good catch.

File: client/blocks/reader-full-post/index.jsx [hunk 2]
```diff
@@ -353,14 +355,38 @@ export class FullPostView extends React.Component {
...
+		if ( feedId ) {
+			props.feed = getFeed( state, feedId );
+		}
+		return props;
+	},
+	dispatch => {
+		return bindActionCreators( {
+			setSection
+		}, dispatch );
+	}
```
Comment (gwwar) [inline client/blocks/reader-full-post/index.jsx]: The second argument can be [shortened](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments) to ```{ setSection }```.

**Comment by bluefuton** [comment]
I've given this a test run with and without new streams enabled, and it looks good to :ship:

**Pull request merged**